pinot-core/src/main/java/org/apache/pinot/core/data/function/DateTimeFunctions.java

```diff
@@ -35,9 +63,79 @@ static Long toEpochHours(Long millis) {
   }
 
   /**
-   * Convert epoch millis to epoch minutes, bucketed by given bucket granularity
+   * Convert epoch millis to epoch hours, bucketed by given bucket granularity
    */
-  static Long toEpochMinutes(Long millis, String bucket) {
-    return TimeUnit.MILLISECONDS.toMinutes(millis) / Integer.parseInt(bucket);
+  static Long toEpochHoursBucket(Long millis, String bucket) {
```

Comment:
The term 'bucket' is not intuitive to me (but that could be just me). You may think of using something like:
```
toEpochHoursRounded(Long millis, String granularity) {
```
with the comment that the method converts the milis to hours, and rounds it down to the nearest granularity specified.

You can decide, though.

Reply:
Agreed, I don't like "bucket" here either, but couldn't think of a better one. Rounded does sound better, but i wonder if it'll be confusing.
For example, consider millis 1578685189000 (2020/01/10 11:39:49)
`toEpochSeconds(millis) = 1578685189`

What we want is - `toEpochSeconds(millis, 10)` = 1578685189/10 = **157868518** i.e. divide the seconds to create **tenSecondsSinceEpoch**
Rounding can be confused with - _keep the same format, but find the nearest 10 minutes value_. 
`toEpochSecondsRounded(millis, 10)` = (1578685189/10) * 10 = **1578685180**

Maybe we should solicit more opinions? Once this function starts being used, it'll be with us forever

Reply:
As per our offline conversation, we now have toEpochXXX, toEpochXXXRounded, toEpochXXXBucket. For example,
1) toEpochSeconds(millis) - converts from millis to  epoch seconds
2) toEpochSecondsRounded (millis, 10) - converts from millis to epoch seconds, and rounds to the smaller 10th value i.e. (seconds/10) * 10
3) toEpochSecondsBucket(millis, 10) - converts from millis to epoch seconds and divides by bucket, to get tenSecondsFromEpoch i.e. seconds / 10